xds/internal/client/client_watchers.go

```diff
@@ -213,7 +268,7 @@ func (c *Client) watch(wi *watchInfo) (cancel func()) {
 func (c *Client) watchLDS(serviceName string, cb func(ListenerUpdate, error)) (cancel func()) {
 	wi := &watchInfo{
 		c:           c,
-		typeURL:     version.V2ListenerURL,
+		typeURL:     c.listenerURL(),
```

Comment:
Should we make an enum type for the types, and use that here? So that `Client` wouldn't need to deal with `v2` and `v3`. The underlying versioned client can do the conversion.

We had this enum at the beginning (I think?), but later found it's not necessary, and we can just use the string. Now with two versions, the enum might make more sense.

Reply:
I think making that change has definitely made the code a little cleaner. But now the change is huge :(